File: src/Test/PolymorphismFFICheatcode.js
class Monoidish {
  mempty_() { return 1; }
  mappend_(x, y) { return x + y; }
}
export const runPolymorphismFFICheatcode = function(limit) {
  let n = Math.floor(limit);
  let acc = 0;
  let m = new Monoidish();
  while (n > 0) {
    acc = m.mappend_(acc, m.mempty_());
    n--;
  }
  return acc;
};

File: src/Test/PolymorphismFFICheatcode.java
    private interface Monoidish {
        int mempty();
        int mappend(int left, int right);
    }

    private static final class IntMonoidish implements Monoidish {
        public int mempty() { return 1; }
        public int mappend(int left, int right) { return left + right; }
    }

    public static final java.util.function.Function<Object, Object> runPolymorphismFFICheatcode = Bench.nativeBenchmark(input -> {
        int count = (Integer) input;
        int result = 0;
        Monoidish dictionary = new IntMonoidish();
        for (int i = 0; i < count; i++) {
            result = dictionary.mappend(result, dictionary.mempty());
        }
        return result;
    });
